### ISSUE TITLE
Accessibility: Address headings on My Account page out of order

### ISSUE THREAD

**Issue by amberhinds** [woothemes/woocommerce#43630]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

The address titles on the My Account Address page are set to H3s. The only other heading on the page is the "My Account" H1, so these skip a level.
https://github.com/woocommerce/woocommerce/blob/758df4854d7b627fc26eb5c0860dc492dea6a06e/plugins/woocommerce/templates/myaccount/my-address.php#L62
<img width="243" alt="Screenshot 2024-01-14 at 9 43 59 PM" src="https://github.com/woocommerce/woocommerce/assets/13153456/2b10ef7c-c6cf-4261-883c-2ea4812456c5">

### Expected behavior

Headings should create a hierarchical outline on the page and not skip levels. The address titles should be coded as H2s not H3s.

### Actual behavior

Heading skip from H1 to H3.

### Steps to reproduce

See code linked above.

### WordPress Environment

N/A

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woo.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by Manussakis** [comment]
I have started working on this task.
